### PR TITLE
Implement AbsolutePathCorrectionEngine

### LINKED ISSUE #99: Create AbsolutePathCorrectionEngine class
**EDIT**:
1. Rectified SuggestionEngine -> CorrectionEngine
2. Added proposed implementation
3. Add gotchas

---

As described in #97, I'll be implementing the `AbsolutePathCorrectionEngine` class.

**Proposed API (I'm only showing the important methods below)**:

```java
class AbsolutePathCorrectionEngine implements CorrectionEngine<AbsolutePath> {
  public AbsolutePathCorrectionEngine(Model model) { }
  public CorrectionResult<AbsolutePath> correct(AbsolutePath inputPath) { } 
}
```

**Proposed Implementation (step by step)**:
1. Get the components of the uncorrected absolute path. For example, the path `/path/to/blo` will have the components `["path", "to", "blo"]`.
2. Compare this uncorrected path component-by-component with a list of possible paths. For every two components, compute their distance and add it to the total sum of the distance between two paths.
3. Return the path with the least total distance.

**Gotchas (based on the current implementation)**:
1. Currently, I'm not prioritizing a longer match over a shorter match. That is, if the input is `/path/to/bloc`, the correction engine might return `/path/to/blo` over `/path/to/block` (if time allows, I'll create another issue to address this)

#### How to use
1. I'd imagine that you instantiate this class during the creation of both `SuggestionEngine` and `Parser`. During this instantiation phase, you'll supply the app's model to `AbsolutePathCorrectionEngine`'s constructor.
2. After that, whenever you receive user input, you'll call the correct method. You're guaranteed to receive a corrected path (or error if `inputPath` is not understandable)

### PR DESCRIPTION
**Pull request by kevinputera**
Closes #99 

This PR works on the following:
- [x] Implements `AbsolutePathCorrectionEngine`
- [x] Write test cases for `AbsolutePathCorrectionEngine`

### CODE REVIEW

**Comment by kevinputera** [comment]
Sorry for the delay @ljiazh3ng @johannagwan , please help review 👀 !!

File: src/main/java/com/notably/logic/correction/AbsolutePathCorrectionEngine.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    private final Model model;
+    private final int distanceThreshold;
+
+    public AbsolutePathCorrectionEngine(Model model, int distanceThreshold) {
+        this(new LevenshteinDistanceCalculator(), model, distanceThreshold);
+    }
+
+    public AbsolutePathCorrectionEngine(EditDistanceCalculator editDistanceCalculator,
+            Model model, int distanceThreshold) {
+        Objects.requireNonNull(editDistanceCalculator);
```
Comment (johannagwan) [inline src/main/java/com/notably/logic/correction/AbsolutePathCorrectionEngine.java:32]: Can I ask why you omit the nullity check for `distanceThreshold`?

**Review by johannagwan — APPROVED** [review]
LGTM!

File: src/main/java/com/notably/logic/correction/AbsolutePathCorrectionEngine.java (hunk 1 above)
Comment (kevinputera) [reply src/main/java/com/notably/logic/correction/AbsolutePathCorrectionEngine.java:32]: `distanceThreshold` is an integer, cannot be null!

**Pull request merged**